airgap/terraform/main.tf

```diff
@@ -0,0 +1,305 @@
+    protocol    = "udp"
+    cidr_blocks = ["10.0.0.0/8"]
+  }
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  tags = {
+    Name = "lh-registry-secgrp-${random_string.random_suffix.id}"
+  }
+}
+
+# Create Public subnet
+resource "aws_subnet" "lh_registry_aws_public_subnet" {
+  vpc_id     = aws_vpc.lh_registry_aws_vpc.id
+  availability_zone = "us-east-1a"
+  cidr_block = "10.0.1.0/24"
+
+  map_public_ip_on_launch = true
+
+  tags = {
+    Name = "lh-registry-public-subnet-${random_string.random_suffix.id}"
+  }
+}
+
+# Create private subnet
+resource "aws_subnet" "lh_registry_aws_private_subnet" {
+  vpc_id     = aws_vpc.lh_registry_aws_vpc.id
+  availability_zone = "us-east-1b"
+  cidr_block = "10.0.2.0/24"
+
+  map_public_ip_on_launch = true
+
+  tags = {
+    Name = "lh-registry-private-subnet-${random_string.random_suffix.id}"
+  }
+}
+
+# Create route table for public subnets
+resource "aws_route_table" "lh_registry_aws_public_rt" {
+  depends_on = [
+    aws_internet_gateway.lh_registry_aws_igw,
+  ]
+
+  vpc_id = aws_vpc.lh_registry_aws_vpc.id
+
+  route {
+    cidr_block = "0.0.0.0/0"
+    gateway_id = aws_internet_gateway.lh_registry_aws_igw.id
+  }
+
+  tags = {
+    Name = "lh-registry-aws-public-rt-${random_string.random_suffix.id}"
+  }
+}
+
+# Create route table for private subnets
+resource "aws_route_table" "lh_registry_aws_private_rt" {
+  depends_on = [
+    aws_internet_gateway.lh_registry_aws_igw
+  ]
+
+  vpc_id = aws_vpc.lh_registry_aws_vpc.id
+
+  route {
+    cidr_block = "0.0.0.0/0"
+    gateway_id = aws_internet_gateway.lh_registry_aws_igw.id
+  }
+
+  tags = {
+    Name = "lh-registry-aws-private-rt-${random_string.random_suffix.id}"
+  }
+}
+
+# Associate public subnet to public route table
+resource "aws_route_table_association" "lh_registry_aws_public_subnet_rt_association" {
+  depends_on = [
+    aws_subnet.lh_registry_aws_public_subnet,
+    aws_route_table.lh_registry_aws_public_rt
+  ]
+
+  subnet_id      = aws_subnet.lh_registry_aws_public_subnet.id
+  route_table_id = aws_route_table.lh_registry_aws_public_rt.id
+}
+
+# Associate private subnet to private route table
+resource "aws_route_table_association" "lh_registry_aws_private_subnet_rt_association" {
+  depends_on = [
+    aws_subnet.lh_registry_aws_private_subnet,
+    aws_route_table.lh_registry_aws_private_rt
+  ]
+
+  subnet_id      = aws_subnet.lh_registry_aws_private_subnet.id
+  route_table_id = aws_route_table.lh_registry_aws_private_rt.id
+}
+
+# Create AWS key pair
+resource "aws_key_pair" "lh_registry_aws_pair_key" {
+  key_name   = format("%s_%s", "lh_registry_aws_key_pair", random_string.random_suffix.id)
+  public_key = file(var.aws_ssh_public_key_file_path)
+}
+
+# Create instance for registry
+resource "aws_instance" "lh_registry_aws_instance" {
+
+  depends_on = [
+    aws_subnet.lh_registry_aws_public_subnet,
+  ]
+
+  availability_zone = "us-east-1a"
```

Comment:
why we hard code zone here?

Reply:
> why we hard code zone here?

Just for simplicity.
Actually the AWS account longhorn-intergration-service-account used to run regression tests can only work well in us-east-1, because the resources limits for longhorn-intergration-service-account are only increased in us-east-1. If we change the regression test to run in other zone, we would encounter errors like "resource limit exceeded".

Reply:
Modified to use a variable with default value "us-east-1".